Button 'Setup tables' causes HTTP Status 500 – Internal Server Error

The error appears when you use the deegree webconsole and press the button 'Setup tables' on an activated 'Feature store' or 'Metadata store' configuration. 
![2024_03_08_0o1_Kleki](https://github.com/deegree/deegree3/assets/74777462/a0ab2ffd-244a-4909-96c6-22abf6901c61)

Once the button is pressed the HTTP Status 500 is presented:
![image](https://github.com/deegree/deegree3/assets/74777462/d68ab6fc-352e-4375-944b-11fea5ea7e43)

The deegree server logs are as following:
```
08-Mar-2024 13:17:30.989 SEVERE [http-nio-8080-exec-3] com.sun.faces.application.view.FaceletViewHandlingStrategy.handleRenderException Error Rendering View[/console/generic/sql.xhtml]
	javax.el.ELException: /console/generic/sql.xhtml @21,59 value="${xmlEditorBean.content}": Error reading [content] on type [org.deegree.console.generic.XmlEditorBean]
		at com.sun.faces.facelets.el.TagValueExpression.getValue(TagValueExpression.java:114)
		at javax.faces.component.ComponentStateHelper.eval(ComponentStateHelper.java:194)
		at javax.faces.component.ComponentStateHelper.eval(ComponentStateHelper.java:182)
		at javax.faces.component.UIOutput.getValue(UIOutput.java:174)
		at javax.faces.component.UIInput.getValue(UIInput.java:291)
		at org.primefaces.util.ComponentUtils.getValueToRender(ComponentUtils.java:122)
		at org.primefaces.util.ComponentUtils.getValueToRender(ComponentUtils.java:82)
		at org.primefaces.extensions.component.codemirror.CodeMirrorRenderer.encodeMarkup(CodeMirrorRenderer.java:112)
		at org.primefaces.extensions.component.codemirror.CodeMirrorRenderer.encodeEnd(CodeMirrorRenderer.java:94)
		at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:924)
		at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1863)
		at javax.faces.render.Renderer.encodeChildren(Renderer.java:176)
		at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:894)
		at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:304)
		at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:114)
		at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:894)
		at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:304)
		at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:114)
		at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:894)
		at com.sun.faces.renderkit.html_basic.HtmlBasicRenderer.encodeRecursive(HtmlBasicRenderer.java:304)
		at com.sun.faces.renderkit.html_basic.GroupRenderer.encodeChildren(GroupRenderer.java:114)
		at javax.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:894)
		at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1856)
		at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1859)
		at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1859)
		at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:456)
		at com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:134)
		at com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:120)
		at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:101)
		at com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:219)
		at javax.faces.webapp.FacesServlet.service(FacesServlet.java:647)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1794)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.base/java.lang.Thread.run(Thread.java:829)
	Caused by: javax.el.ELException: Error reading [content] on type [org.deegree.console.generic.XmlEditorBean]
		at javax.el.BeanELResolver.getValue(BeanELResolver.java:100)
		at com.sun.faces.el.DemuxCompositeELResolver._getValue(DemuxCompositeELResolver.java:176)
		at com.sun.faces.el.DemuxCompositeELResolver.getValue(DemuxCompositeELResolver.java:203)
		at org.apache.el.parser.AstValue.getValue(AstValue.java:168)
		at org.apache.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:189)
		at com.sun.faces.facelets.el.TagValueExpression.getValue(TagValueExpression.java:109)
		... 52 more
	Caused by: java.lang.NullPointerException
		at java.base/java.io.File.<init>(File.java:278)
		at org.deegree.console.generic.XmlEditorBean.getContent(XmlEditorBean.java:171)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at javax.el.BeanELResolver.getValue(BeanELResolver.java:95)
		... 57 more
```

The error was first detected by trying to reproduce the bug described in https://github.com/deegree/deegree3/issues/369 

**Affected versions: 3.5.5, 3.5.6**

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/Config.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/Config.java
@@ -57,8 +57,7 @@ import org.slf4j.Logger;
  * @author <a href="mailto:schmitz@lat-lon.de">Andreas Schmitz</a>
  * @since 3.4
  */
-@Named
-@RequestScoped
+
 public class Config implements Comparable<Config>, Serializable {
 
 	private static final long serialVersionUID = -175529275940063759L;

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/connection/sql/ConnectionTester.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/connection/sql/ConnectionTester.java
@@ -62,8 +62,7 @@ import java.io.Serializable;
 public class ConnectionTester implements Serializable {
 
 	private Workspace getWorkspace() {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		return ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace().getNewWorkspace();
+		return WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 	}
 
 	public void test() {

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/connection/sql/JdbcBean.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/connection/sql/JdbcBean.java
@@ -205,8 +205,7 @@ public class JdbcBean implements Serializable {
 
 	private void create() {
 		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		Workspace ws = ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace()
-			.getNewWorkspace();
+		Workspace ws = WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 		try {
 			Map<String, Object> sMap = ctx.getSessionMap();
 			String newId = (String) sMap.get("newConfigId");

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/feature/FeatureStoreConfig.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/feature/FeatureStoreConfig.java
@@ -71,8 +71,7 @@ public class FeatureStoreConfig extends Config {
 	}
 
 	private Workspace getWorkspace() {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		return ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace().getNewWorkspace();
+		return WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 	}
 
 	public boolean getSql() {

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/feature/MappingWizardSQL.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/feature/MappingWizardSQL.java
@@ -48,7 +48,10 @@ import java.util.List;
 import java.util.SortedSet;
 import java.util.TreeSet;
 
+import jakarta.el.ELContext;
+import jakarta.el.ELResolver;
 import jakarta.enterprise.context.SessionScoped;
+import jakarta.faces.application.Application;
 import jakarta.faces.application.FacesMessage;
 import jakarta.faces.context.ExternalContext;
 import jakarta.faces.context.FacesContext;
@@ -133,8 +136,7 @@ public class MappingWizardSQL implements Serializable {
 	}
 
 	private DeegreeWorkspace getWorkspace() {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		return ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace();
+		return WorkspaceBean.getInstance().getActiveWorkspace();
 	}
 
 	public SortedSet<String> getAvailableJdbcConns() {
@@ -279,9 +281,7 @@ public class MappingWizardSQL implements Serializable {
 	public String generateConfig() {
 
 		try {
-			ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-			Workspace ws = ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace()
-				.getNewWorkspace();
+			Workspace ws = WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 
 			CRSRef storageCrs = CRSManager.getCRSRef(this.storageCrs);
 			boolean createBlobMapping = storageMode.equals("hybrid") || storageMode.equals("blob");
@@ -336,9 +336,7 @@ public class MappingWizardSQL implements Serializable {
 
 	public String createTables() throws ClassNotFoundException, SecurityException, NoSuchMethodException,
 			IllegalArgumentException, IllegalAccessException, InvocationTargetException {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		Workspace ws = ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace()
-			.getNewWorkspace();
+		Workspace ws = WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 		FeatureStore fs = ws
 			.init(new DefaultResourceIdentifier<FeatureStore>(FeatureStoreProvider.class, getFeatureStoreId()), null);
 		SQLFeatureStore store = (SQLFeatureStore) fs;
@@ -368,7 +366,7 @@ public class MappingWizardSQL implements Serializable {
 	public String activateFS() {
 		try {
 			ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-			WorkspaceBean workspaceBean = (WorkspaceBean) ctx.getApplicationMap().get("workspace");
+			WorkspaceBean workspaceBean = WorkspaceBean.getInstance();
 			Workspace ws = workspaceBean.getActiveWorkspace().getNewWorkspace();
 			WorkspaceUtils.reinitializeChain(ws,
 					new DefaultResourceIdentifier<FeatureStore>(FeatureStoreProvider.class, getFeatureStoreId()));

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/metadata/MetadataStoreConfig.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/metadata/MetadataStoreConfig.java
@@ -37,9 +37,6 @@ package org.deegree.console.datastore.metadata;
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 
-import jakarta.el.ELContext;
-import jakarta.el.ELResolver;
-import jakarta.faces.application.Application;
 import jakarta.faces.application.FacesMessage;
 import jakarta.faces.component.html.HtmlCommandButton;
 import jakarta.faces.context.FacesContext;

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/metadata/MetadataStoreConfig.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/datastore/metadata/MetadataStoreConfig.java
@@ -37,9 +37,11 @@ package org.deegree.console.datastore.metadata;
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 
+import jakarta.el.ELContext;
+import jakarta.el.ELResolver;
+import jakarta.faces.application.Application;
 import jakarta.faces.application.FacesMessage;
 import jakarta.faces.component.html.HtmlCommandButton;
-import jakarta.faces.context.ExternalContext;
 import jakarta.faces.context.FacesContext;
 import jakarta.faces.event.ActionEvent;
 
@@ -63,8 +65,7 @@ public class MetadataStoreConfig extends Config {
 	}
 
 	private Workspace getWorkspace() {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		return ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace().getNewWorkspace();
+		return WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 	}
 
 	public void updateId(ActionEvent evt) {

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/generic/Connection.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/generic/Connection.java
@@ -62,8 +62,7 @@ public class Connection implements Serializable {
 	private String status = "OK";
 
 	private Workspace getWorkspace() {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		return ((WorkspaceBean) ctx.getApplicationMap().get("workspace")).getActiveWorkspace().getNewWorkspace();
+		return WorkspaceBean.getInstance().getActiveWorkspace().getNewWorkspace();
 	}
 
 	public String getId() {

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/generic/XmlEditorBean.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/generic/XmlEditorBean.java
@@ -188,14 +188,17 @@ public class XmlEditorBean implements Serializable {
 		if (content == null) {
 			LOG.trace("No content set for {}", this.toString());
 			if (resourceProviderClass == null) {
-				File file = getFile();
-				if (fileName != null && file.exists()) {
-					LOG.trace("Loading content from file: {}", file.getAbsolutePath());
-					content = FileUtils.readFileToString(file);
-					LOG.trace("Setting content to: {}", content);
-					return content;
+				if (fileName != null) {
+					File file = getFile();
+					if (file.exists()) {
+						LOG.trace("Loading content from file: {}", file.getAbsolutePath());
+						content = FileUtils.readFileToString(file);
+						LOG.trace("Setting content to: {}", content);
+						return content;
+					}
 				}
-				else if (emptyTemplate != null) {
+
+				if (emptyTemplate != null) {
 					// load template content if the requested file did not exists
 					LOG.trace("Loading template from {}", emptyTemplate);
 					StringWriter sw = new StringWriter();

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/moduleinfo/ModulesBean.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/moduleinfo/ModulesBean.java
@@ -76,8 +76,7 @@ public class ModulesBean implements Serializable {
 	}
 
 	public List<String> getWorkspaceModules() {
-		ExternalContext ctx = FacesContext.getCurrentInstance().getExternalContext();
-		WorkspaceBean wsBean = ((WorkspaceBean) ctx.getApplicationMap().get("workspace"));
+		WorkspaceBean wsBean = WorkspaceBean.getInstance();
 		if (wsBean == null) {
 			return Collections.emptyList();
 		}

--- a/deegree-services/deegree-webservices/src/main/java/org/deegree/console/workspace/WorkspaceBean.java
+++ b/deegree-services/deegree-webservices/src/main/java/org/deegree/console/workspace/WorkspaceBean.java
@@ -54,7 +54,10 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 
+import jakarta.el.ELContext;
+import jakarta.el.ELResolver;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.faces.application.Application;
 import jakarta.faces.application.FacesMessage;
 import jakarta.faces.context.FacesContext;
 
@@ -426,6 +429,16 @@ public class WorkspaceBean implements Serializable {
 		this.uploadWorkspace();
 		this.unzipWorkspace();
 		LOG.debug("Workspace {} unzipped into workspace root directory", workspaceImportName);
+	}
+
+	public static WorkspaceBean getInstance() {
+		FacesContext facesContext = FacesContext.getCurrentInstance();
+		Application application = facesContext.getApplication();
+
+		ELResolver elResolver = application.getELResolver();
+		ELContext elContext = facesContext.getELContext();
+
+		return (WorkspaceBean) elResolver.getValue(elContext, null, "workspace");
 	}
 
 }

--- a/deegree-services/deegree-webservices/src/main/webapp/console/generic/sql.xhtml
+++ b/deegree-services/deegree-webservices/src/main/webapp/console/generic/sql.xhtml
@@ -7,18 +7,13 @@
         <legend>
           <h:outputText styleClass="titel" value="Setup database" />
         </legend>
-        <pe:codeMirror id="sqlArea" mode="sql" theme="dracular"
-                       value="#{execution.statements}" keyMap="debug" lineNumbers="true"
-                       matchBrackets="true"/>
-        <h:inputTextarea id="sqlArea" cols="70" rows="21" value="" />
+        <h:inputTextarea id="sqlArea" cols="70" rows="21" value="#{execution.statements}" />
         <h:outputText value="#{execution.message}" />
         <br /> <br />
         <h:commandButton styleClass="buttonSave" value="#{labels.sql_execute}" action="#{execution.execute}" />
         <h:commandButton styleClass="buttonCancel" value="#{labels.sql_back}" action="#{execution.getBackOutcome}" />
         <p id="highlighting" style="cursor: pointer;" class="schema" onclick="activateEditor();">Turn on
           highlighting</p>
-        <pe:codeMirror id="xmltextarea" mode="xml" theme="dracular" lineNumbers="true" matchBrackets="true"
-                       value="${xmlEditorBean.content}" />
       </fieldset>
     </h:form>
   </ui:define>

